api/proto/teleport/legacy/client/proto/joinservice.proto

```diff
@@ -194,6 +194,119 @@ message RegisterUsingOracleMethodResponse {
   }
 }
 
+// RegisterUsingBoundKeypairInitialRequest is the initial request sent during
+// the join process for bound-keypair joining.
+message RegisterUsingBoundKeypairInitialRequest {
+  // Registration parameters common to all join methods.
+  types.RegisterUsingTokenRequest join_request = 1;
+
+  // If set, attempts to bind a new keypair using an initial join secret.
+  // Any value set here will be ignored if a keypair is already bound.
+  string initial_join_secret = 2;
```

Comment:
minor nit/no strong opinions: should these be `optional` fields?

Reply:
Ah, good call. Made the change for this field, and removed `new_public_key` outright.

Reply:
Bah. This file is built using protoc-gen-gogo which doesn't support field presence, so this just generates errors. I'm unfortunately reverting this change and we'll just stick to empty/nonempty for now :(